### PR TITLE
Stop unnecessary unserializable entity computation

### PR DESCRIPTION
**Pull request by namanjain**
For parallel processing mode, unserializable entities don't need to be
computed inside the main process unless returned. This change stops the
current unnecessary computation.

### CODE REVIEW

File: bionic/deriver.py [hunk 1]
```diff
@@ -459,29 +461,38 @@ def _process_entry(self, entry, task_key_logger):
         # Initialize the task state before attempting to complete it.
         state.initialize(self._bootstrap, self._flow_instance_uuid)
 
-        # self._bootstrap.process_executor is None when parallel processing is
-        # disabled or when entities are bootstrapped.
         if (
-            not state.provider.attrs.should_persist()
+            # Bootstrap entity.
```
Comment (jqmp) [inline bionic/deriver.py:465]: Nit: can we make these complete sentences, like "This is a bootstrap entity."?

File: bionic/task_state.py [hunk 2]
```diff
@@ -41,43 +41,42 @@ def __init__(self, task, dep_states, case_key, provider):
...
-
-    @property
-    def is_completable(self):
+    # consider making the way it computes in parallel mode the default.
+    def blocking_dep_states(self):
         """
-        Indicates whether the task state can be completed in it's current shape.
-        True when the state is initialized, all of its persisted dependencies are
-        completed, and all its other dependencies are computable.
+        Returns all of it's persisted dependencies that aren't yet completed and
```
Comment (jqmp) [inline bionic/task_state.py:47]: s/it's/its/ -- or better yet, /this task state's/.

File: bionic/task_state.py [hunk 3]
```diff
@@ -41,43 +41,42 @@ def __init__(self, task, dep_states, case_key, provider):
...
+        return set(
+            blocking_dep_state_tk
+            for blocking_dep_state in blocking_dep_states
+            for blocking_dep_state_tk in blocking_dep_state.task.keys
         )
 
+    @property
+    def is_completable(self):
+        """
+        Indicates whether the task state can be completed in it's current shape.
```
Comment (jqmp) [inline bionic/task_state.py:75]: s/it's/its/

File: bionic/deriver.py [hunk 4]
```diff
@@ -459,29 +461,38 @@ def _process_entry(self, entry, task_key_logger):
...
             # a tmp file for this instead to use protocol for serialization instead of
             # using cloudpickle.
             state.complete(task_key_logger)
             self._mark_entry_completed(entry)
-        else:
-            # NOTE 1: Logging support for multiple processes not done yet.
+
+        # Process serializable entity in parallel.
+        elif state.provider.attrs.should_persist():
+            # NOTE: Logging support for multiple processes not done yet.
```
Comment (jqmp) [inline bionic/deriver.py:484]: Would it make sense for these `NOTE`s to be `TODO`s?

Comment (namanjain) [reply bionic/deriver.py:484]: Yeah, makes sense to me.

**Comment by namanjain** [comment]
> Makes sense to me! I hope we can come back to the definitions of "blocking", "completable", and "complete" at some point though. :)

Definitely, that's why I left a TODO :)

**Pull request merged**